org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/SwitchStatement.java

```diff
@@ -978,7 +978,7 @@ public void resolve(BlockScope upperScope) {
 								IntPredicate check = (idx) -> {
 									Constant c2 = this.otherConstants[idx].c;
 									if (con.typeID() == TypeIds.T_JavaLangString) {
-										return c2.stringValue().equals(con.stringValue());
+										return c.intValue() == this.otherConstants[idx].intValue();
```

Comment:
That shows an Error if two Strings have the same hashCode. For exampe "FB" and "Ea" gives "Duplicate case".